### ISSUE TITLE
Version `rejectAll()` documentation is basically non-existant

### ISSUE THREAD

**Issue by toadzky** [gradle/gradle#30350]
### Issue type

Missing information

### Problem description

I'm trying to write a catalog that allows a consumer to opt-in to a `rejectAll` version but it fails with a `Could not resolve group:artifact:{reject all versions}.` There's no documentation on how to correctly use `rejectAll` and how it interacts with other things.

### Page with the problem

https://docs.gradle.org/current/userguide/platforms.html

**Comment by ljacomet** [comment]
This is a valid documentation issue.

***

Documentation is indeed lacking, especially as this value is the only `boolean` one in the version catalog.
It should be something like:
`commons-lang3 = { group = "org.apache.commons", name = "commons-lang3", version = { rejectAll = true } }`

@toadzky Can you try the above, adapted to your needs? Could you also share the snippet and the stacktrace at the origin of your issue?

**Comment by toadzky** [comment]
I have defined some `rejectAll` items in my catalog. Here's what happens when I try to include one as a dependency:

```* What went wrong:
Execution failed for task ':module:compileKotlin'.
> Could not resolve all files for configuration ':module:compileClasspath'.
   > Could not resolve software.amazon.awssdk:apache-client:{reject all versions}.
     Required by:
         project :module
      > Module 'software.amazon.awssdk:apache-client' has been rejected:
           Dependency path 'project:module:unspecified' --> 'software.amazon.awssdk:apache-client:{reject all versions}'
```

It's unclear in the documentation how to _use_ the rejected dependency. It's relatively clear how to _define_ a rejected dependency, but not how to use it.

**Comment by toadzky** [comment]
Oh, you asked for some snippets.... 

My catalog is kotlin-defined: 

```kotlin 
library("aws-sdk-client-reject-apache", "software.amazon.awssdk", "apache-client").version {
  rejectAll()
}
```

the generated toml look like this:

```toml
aws-sdk-client-reject-apache = {group = "software.amazon.awssdk", name = "apache-client", version = { rejectAll = true } }
```

and the usage i have is 

```kotlin
  implementation(libs.aws.sdk.client.reject.apache)
```

i'm not sure what stacktrace you would like me to send you, i don't see one in the gradle output.

**Comment by toadzky** [comment]
yeah, i don't really understand what you mean by doing it as a constraint. it's literally a version setting for dependencies, but you are saying it doesn't make sense to use it in the dependency section. this is basically the problem: it's completely unclear how this functionality is supposed to be used, the docs just show how to set the dependency version to be rejected.

**Comment by ljacomet** [comment]
Sorry, should have added a link to the documentation: https://docs.gradle.org/current/userguide/dependency_constraints.html#sec:adding-constraints-transitive-deps
